### PR TITLE
common: shell: Add shell power commands

### PR DESCRIPTION
**Pull request by MouchenHung-QUANTA**
Summary:
- Add shell power commands
  - Get power status
    --> platform power status
  - Control power
    --> platform power control [device_name] [status]
    * **[device_name]**: host/bmc
    * **[status]**: on/off/reset

Test Plan:
- Build Code: PASS
- Check command results in BIC console: PASS

Log:
- BIC console:
  ```
  uart:~$ platform power status
  ------------------------------------
  [HOST]
  * status:      on
  * cpu status:  on
  * post status: on
  [BMC]
  * status: non-support
  ------------------------------------

  uart:~$ platform power control host off
  HOST power off successed!

  uart:~$ platform power status
  ------------------------------------
  [HOST]
  * status:      off
  * cpu status:  off
  * post status: off
  [BMC]
  * status: non-support
  ------------------------------------
  ```

### CODE REVIEW

File: common/shell/commands/power_shell.c [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+			shell_warn(shell, "Selected device not support %s", argv[2]);
+			return;
+		}
+
+		if (check_pwr_ctl_valid(shell, control_opt, get_DC_status()))
+			return;
+
+		int ret = pal_host_power_control(control_opt);
+		if (ret == 0)
+			shell_print(shell, "%s power %s successed!", device_info[DEVICE_HOST].name,
```
Comment (GoldenBug) [inline common/shell/commands/power_shell.c:125]: > succeeded

File: common/shell/commands/power_shell.c [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+		dev_idx = DEVICE_HOST;
+	} else if (!strcmp(argv[1], BMC_KEYWORD)) {
+		dev_idx = DEVICE_BMC;
+	} else {
+		shell_warn(shell, "Uknown device type %s", argv[1]);
+		return;
+	}
+
+	if (device_info[dev_idx].enable == DISABLE) {
+		shell_warn(shell, "%s power control is not support", device_info[dev_idx].name);
```
Comment (GoldenBug) [inline common/shell/commands/power_shell.c:96]: > supported

File: common/shell/commands/power_shell.c [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+	if (argc != 1) {
+		shell_warn(shell, "Help: platform power status");
+		return;
+	}
+
+	shell_print(shell, "------------------------------------");
+	for (int i = 0; i < MAX_DEVICE_CNT; i++) {
+		if (device_info[i].enable == DISABLE) {
+			shell_print(shell, "[%s]", device_info[i].name);
+			shell_print(shell, "* status: non-support", device_info[i].name);
```
Comment (GoldenBug) [inline common/shell/commands/power_shell.c:56]: > not supported

File: common/shell/commands/power_shell.h [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+struct device_arr {
+	uint8_t enable;
+	char *name;
+};
+
+enum device_id {
+	DEVICE_HOST,
+	DEVICE_BMC,
+	MAX_DEVICE_CNT,
```
Comment (GoldenBug) [inline common/shell/commands/power_shell.h:37]: extend to `MAX_DEVICE_COUNT`

**Review by GoldenBug — CHANGES_REQUESTED** [review]
A couple of minor change requests. Otherwise looks great!

**Comment by MouchenHung-QUANTA** [comment]
Hi @GoldenBug , new commit has been uploaded, please have a look, thanks!

**Pull request closed without merging**